client/driver/executor/executor.go

```diff
@@ -21,6 +23,18 @@ import (
 	"github.com/hashicorp/nomad/nomad/structs"
 )
 
+// Executor is the interface which allows a driver to launch and supervise
+// a process
+type Executor interface {
+	LaunchCmd(command *ExecCommand, ctx *ExecutorContext) (*ProcessState, error)
+	LaunchSyslogServer(ctx *ExecutorContext) (*SyslogServerState, error)
+	Wait() (*ProcessState, error)
+	ShutDown() error
+	Exit() error
+	UpdateLogConfig(logConfig *structs.LogConfig) error
```

Comment:
I would add a // COMPAT: prior to Nomad 0.3.2, UpdateTask did not exist. This is so we can grep the code base for COMPAT and know what to remove eventually